### DISCUSSION TITLE
Sync GitHub Issues with GitButler Virtual Branches

### DISCUSSION THREAD

**Discussion by noamsiegel** [gitbutlerapp/gitbutler#5377] [Ideas]
I'd like to propose and contribute to a feature that would sync GitHub repository issues with GitButler virtual branches. This integration could significantly improve workflow efficiency for developers using both platforms.

I can work on this myself, but I need these questions answered first:

- Is this feature aligned with GitButler's roadmap?
- What would be the best approach to implement this integration?
- Are there any existing efforts or discussions around this topic?

I'd like to have a veteran walk me through (either on a call or a recording) these questions so that I can contribute as soon as possible. Or, a well-detailed explanation in the comments below would suffice!

**Comment by Byron** [comment]
Hi @noamsiegel , thanks for asking!

Without my answer being authoritative at all, I'd think anything that improves the efficiency of its users is aligned with GitButler - it's here to serve :).

Before discussing specifics, I think it would be great to learn from you how you would use such a feature, in enough detail to understand how it would work. For instance, taking the proposition literally my GitButler would now have 243 lanes, one for each issue, which probably isn't what you had in mind - I am clearly missing something. Everything you can write to help others align with your initial vision would be greatly appreciated.

Let me also CC @krlvi who is always interested in improved workflows as well.

**Reply by noamsiegel** [reply]
Ah, I was hoping to have another tab like "Target" and "Workspace" that would be called "Issues." This would have a modular filter that could be narrowed down to:

- Issue status/type
- Assignees and authors
- Labels and milestones
- Date and time

Switching between repositories would be the **first filter** for issues that can be created into lanes. 

I completely understand your point about having too many issues all at once. That is why, on this "issues" tab, there would be a way to turn an issue into a lane. That would, in turn, tie that lane's PR to that issue.

Let me know if that makes sense. I don't want 100+ lanes all of a sudden, too!

**Reply by noamsiegel** [reply]
As an aside, the UI element should be reviewed heavily before implementation. For example, It could also stand in the same column layout as the "Branches" — instead of my aforementioned suggestion under "Target" and "Workspaces".

**Reply by Byron** [reply]
Thanks for elaborating! This sounds quite straightforward to me, so straightforward in fact that I'd think somebody like @PavelLaptev might have thought about it already.

In any case, this is probably it from me for the first gathering and clarification of the issue at hand, let's see when others chime in to figure out if it makes sense to tackle this.

**Reply by noamsiegel** [reply]
No problem! I'll wait for other opinions/responses in the meantime.

**Reply by noamsiegel** [reply]
Any update on this @Byron?

**Reply by PavelLaptev** [reply]
@noamsiegel Thanks for the idea! Right now, we're working on the [design for V3](https://www.figma.com/design/ShIiR6hI5dzH7sT5L03Jg1/Design-3.0?node-id=748-162943&t=j8DywzVjahN06Cnd-1), where the "branches" sidebar will move to a separate page, so the classic sidebar will go away. 

There could be a separate page for "issues" too, but that's not a priority at the moment. My suggestion is to keep this idea for later, once we have a better understanding of V3. Prototypes or concepts are always welcome, though they wouldn’t be implemented yet as the app is currently in a transition phase. _@krlvi, feel free to correct me._

**Reply by noamsiegel** [reply]
@PavelLaptev This sounds great; I'll keep this on the back burner until V3 is released. Thanks for the update!
